src/app/components/containers/products-details/products-details.component.ts

```diff
@@ -1,25 +1,65 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, OnInit, Output } from '@angular/core';
 import { productsMockList } from '../../mocks/products.mock';
-import { Product } from '../../../modules/shared/types/product';
+import { OrderProduct } from '../../../modules/shared/types/product.order';
 import { Observable } from 'rxjs/internal/Observable';
 import { ActivatedRoute } from '@angular/router';
-import { of } from 'rxjs';
+import { interval, of } from 'rxjs';
+import { ShoppingCartService } from 'src/app/modules/shared/services/shopping.cart.service';
+import { HttpClient } from '@angular/common/http';
+import { ProductDto } from 'src/app/modules/shared/types/product.dto';
+import { environment } from 'src/environments/environment';
+import { AppNavigationService } from 'src/app/modules/shared/services/app-navigation.service';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { ProductService } from 'src/app/services/products.service';
+
+@UntilDestroy({})
 @Component({
   selector: 'app-products-details',
   templateUrl: './products-details.component.html',
   styleUrls: ['./products-details.component.scss'],
 })
 export class ProductsDetailsComponent implements OnInit {
-  product: Product = productsMockList[0];
+  private shoppingCartService: ShoppingCartService;
+  product: ProductDto = {
+    category: '',
+    name: '',
+    price: 0,
+    image: '',
+    id: '',
+    description: '',
+  };
 
-  constructor(private route: ActivatedRoute) {}
-  getProduct(id: String): Observable<Product> {
-    const product = productsMockList.find((product) => product.id === id)!;
-    return of(product);
+  constructor(
+    private productService: ProductService,
+    private route: ActivatedRoute,
+    shoppingCartService: ShoppingCartService,
+    private http: HttpClient,
+    private appNavigationService: AppNavigationService
+  ) {
+    this.shoppingCartService = shoppingCartService;
+  }
+  onNavigateToProductList() {
+    this.appNavigationService.navigateToProductsList();
+  }
+  onNavigateToProductEdit(id: string) {
+    this.appNavigationService.navigateToProductEdit(id);
   }
-
   ngOnInit(): void {
```

Comment:
Would be nice to keep an order for the methods, like having the ngOnInit lifecycle hook after the constructor and then have the other methods.